### PR TITLE
ACA-113 - Submit Noc Request Event & Apply NoC Decision integration

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ACA-113

### Change description ###

Link Apply NocDecision callback with Submit Noc Request Event

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by mattnayler — CHANGES_REQUESTED** [review]
LGTM.  Just a few minor changes: some that can wait until the AAT orgs are created.

File: aca-docker/mocks/wiremock/__files/prd_users_template_Org-Two.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    },
+    {
+      "userIdentifier": "",
+      "firstName": "Tony",
+      "lastName": "Solicington",
+      "email": "befta.master.solicitor.tony@gmail.com",
+      "idamStatus": "ACTIVE"
+    }
+  ],
+  "organisationIdentifier": "Org-Two"
```
Comment (mattnayler) [inline aca-docker/mocks/wiremock/__files/prd_users_template_Org-Two.json:18]: NB: all Org-Two references will need to be updated once the AAT group is created. Although we may as well keep the file names so they match confluence.

File: aca-docker/mocks/wiremock/__files/prd_users_Org-Two.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (mattnayler) [inline aca-docker/mocks/wiremock/__files/prd_users_Org-Two.json:1]: Please delete this file as well as exclude it from git via `.gitignore`

**Review by mattnayler — CHANGES_REQUESTED** [review]
Additional review comments added

File: aca-docker/bin/env_variables_all.txt [hunk 3]
```diff
@@ -13,3 +13,8 @@ export CCD_BEFTA_SOLICITOR_4_PWD=Pa55word11
 export ES_ENABLED_DOCKER=true
```
Comment (mattnayler) [inline aca-docker/bin/env_variables_all.txt:13]: Delete this if it is no longer required

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -45,7 +44,11 @@ def secrets = [
     secret('microservicekey-xui-webapp', 'BEFTA_S2S_CLIENT_SECRET_OF_XUI_WEBAPP'),
     secret('microservicekey-ccd-data', 'BEFTA_S2S_CLIENT_SECRET_OF_CCD_DATA'),
     secret('microservicekey-aac-manage-case-assignment', 'BEFTA_S2S_CLIENT_SECRET_OF_AAC_MANAGE_CASE_ASSIGNMENT')
-  ]
+  ],
+   'aac-${env}': [
+      secret('idam-mca-system-user-username', 'BEFTA_IDAM_CAA_USERNAME'),
+      secret('idam-mca-system-user-password', 'BEFTA_IDAM_CAA_PASSWORD')
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply Jenkinsfile_CNP]: Addressed in https://github.com/hmcts/aac-manage-case-assignment/pull/191/commits/1839013564d73fe8371d1af428ccf78dfadfc25a

File: aca-docker/bin/findPrdUserIds.sh [hunk 5]
```diff
@@ -21,36 +21,41 @@ DB_NAME=openidm
...
+  echo $OUTPUT_FILE | jq . > "$2"
+  echo "${RED}${BOLD}*** Values in $2 have been updated ***${NORMAL}"
+  echo ""
+}
 
-echo ""
-echo "${RED}${BOLD}*** Values in prd_users.json have been updated ***"
-echo -e "*** Please restart your aca-wiremock docker container for changes to take effect ***${NORMAL}"
+generatePrdUsersFileFromTemplate "prd_users_template.json" "prd_users.json"
+generatePrdUsersFileFromTemplate "prd_users_template_Org-Two.json" "prd_users_Org-Two.json"
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply aca-docker/bin/findPrdUserIds.sh:61]: Readme was already updated to list the files,  have added a comment in the script in https://github.com/hmcts/aac-manage-case-assignment/pull/191/commits/1839013564d73fe8371d1af428ccf78dfadfc25a

File: aca-docker/mocks/wiremock/mappings/organisation-users-ORG-2.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
```
[the remark this answers is not in the document]
Comment (sullivangareth) [reply aca-docker/mocks/wiremock/mappings/organisation-users-ORG-2.json]: Renamed in https://github.com/hmcts/aac-manage-case-assignment/pull/191/commits/1839013564d73fe8371d1af428ccf78dfadfc25a

File: aca-docker/mocks/wiremock/__files/prd_users_template_Org-Two.json (hunk 1 above)
Comment (sullivangareth) [reply aca-docker/mocks/wiremock/__files/prd_users_template_Org-Two.json:18]: Agreed - I have used `Org-Two` as this was the org id used in feature tests - we can rename accordingly when orgs are created/assigned

File: aca-docker/mocks/wiremock/__files/prd_users_Org-Two.json (hunk 2 above)
Comment (sullivangareth) [reply aca-docker/mocks/wiremock/__files/prd_users_Org-Two.json]: I excluded it via `.gitignore`... but I did it after commiting the file 🤦 

File deleted

File: aca-docker/bin/env_variables_all.txt (hunk 3 above)
Comment (sullivangareth) [reply aca-docker/bin/env_variables_all.txt]: Good spot - removed in https://github.com/hmcts/aac-manage-case-assignment/pull/191/commits/1839013564d73fe8371d1af428ccf78dfadfc25a

File: aca-docker/bin/findPrdUserIds.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (sullivangareth) [reply aca-docker/bin/findPrdUserIds.sh:61]: Lets see what the 2nd Org looks like when its created and we can rename everything consistently

[the remark this answers is not in the document]
Comment (mattnayler) [reply aca-docker/bin/findPrdUserIds.sh:61]: I am tempted to say we should go with file names including Org-One, Org-Two, Org-Three (i.e. to match confluence) and just use the 'real' AAT Organisation IDs  inside the JSON.

File: build.gradle [hunk 7]
```diff
@@ -286,7 +286,7 @@ dependencies {
   integrationTestImplementation sourceSets.main.runtimeClasspath
   integrationTestImplementation sourceSets.test.runtimeClasspath
 
-  testImplementation group: 'uk.gov.hmcts', name: 'befta-fw', version: '6.2.0'
+  testImplementation group: 'uk.gov.hmcts', name: 'befta-fw', version: '6.5.3-ACA-113'
```
Comment (mattnayler) [inline build.gradle:289]: This will need updating when the following `befta-fw` updates are merged, see hmcts/befta-fw#128.

File: build.gradle [hunk 8]
```diff
@@ -286,7 +289,7 @@ dependencies {
   integrationTestImplementation sourceSets.main.runtimeClasspath
   integrationTestImplementation sourceSets.test.runtimeClasspath
 
-  testImplementation group: 'uk.gov.hmcts', name: 'befta-fw', version: '6.2.0'
+  testImplementation group: 'uk.gov.hmcts', name: 'befta-fw', version: '6.5.4-ACA-113'
```
Comment (mattnayler) [inline build.gradle:292]: This will need updating when the following befta-fw updates are merged, see hmcts/befta-fw#128.

**Review by smathangi — APPROVED** [review]
LGTM

File: aca-docker/bin/users.json [hunk 9]
```diff
@@ -1,4 +1,6 @@
 [
   {"email": "master.caa@gmail.com", "roles": "caseworker,caseworker-caa", "lastName": "caa", "firstName": "master"},
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply aca-docker/bin/users.json:2]: I though `prd-admin` was a temporary thing (though happy to keep them in line with AAT).

Can you confirm if `prd-admin` was added to `master.caa@gmail.com` or `noc.approver@gmail.com` (as I had assumed ACA-106 used the approver account)?

[the remark this answers is not in the document]
Comment (smathangi) [reply aca-docker/bin/users.json:2]: sorry.. my bad "noc.approver@gmail.com".  It's not a temporary fix..PRD wants this way until they restructure their access mechanism.

**Review by smathangi — APPROVED** [review]

File: build.gradle [hunk 10]
```diff
@@ -286,7 +290,7 @@ dependencies {
   integrationTestImplementation sourceSets.main.runtimeClasspath
   integrationTestImplementation sourceSets.test.runtimeClasspath
 
-  testImplementation group: 'uk.gov.hmcts', name: 'befta-fw', version: '6.2.0'
+  testImplementation group: 'uk.gov.hmcts', name: 'befta-fw', version: '6.10.1-ACA-113'
```
Comment (mattnayler) [inline build.gradle:293]: This will need updating when the following befta-fw updates are merged, see hmcts/befta-fw#128.

File: build.gradle [hunk 11]
```diff
@@ -286,7 +290,7 @@ dependencies {
   integrationTestImplementation sourceSets.main.runtimeClasspath
   integrationTestImplementation sourceSets.test.runtimeClasspath
 
-  testImplementation group: 'uk.gov.hmcts', name: 'befta-fw', version: '6.2.0'
+  testImplementation group: 'uk.gov.hmcts', name: 'befta-fw', version: '6.11.1-ACA-113'
```
Comment (mattnayler) [inline build.gradle:293]: This will need updating when the following befta-fw updates are merged, see hmcts/befta-fw#128.

**Review by smathangi — APPROVED** [review]
LGTM

**Review by mattnayler — APPROVED** [review]

**Pull request merged**